packages/components/src/ui/form-group/form-group-label.js

```diff
@@ -18,7 +18,7 @@ function FormGroupLabel( { children, id, labelHidden = false, ...props } ) {
 
 	if ( labelHidden ) {
 		return (
-			<VisuallyHidden as="label" htmlFor={ id }>
+			<VisuallyHidden as="label" htmlFor={ id?.toString() }>
```

Comment:
Curious why this change was needed?

Reply:
The previous prop types were not comprehensive so they didn't have strict checking of htmlFor. But `id` comes in as either a `number | string` and `htmlFor`, when properly type checked against a `label` only accepts strings.

The polymorphism that existed in the previous component implementation didn't actually make any difference to the types, but now that it's a true `PolymorphicComponent` that has `ViewOwnProps` for its arguments, it actually gets the correct full prop types when using the `as` prop.

Reply:
Question though, it's just for typing right? I mean, a user passing a numeric value there will still work? (thinking about BC)

Reply:
Yup! It'll still work, just a type issue.